modules/assim.sequential/inst/NEFI/generate.gefs.xml.R

```diff
@@ -0,0 +1,135 @@
+if (!is.na(args[2])) {
+  source_date <- args[2]
+}
+
+# These lines from the PEcAn workflow load the settings object.
+settings <- PEcAn.settings::read.settings(args[1])
+
+xmloutdir <- regmatches(args[1], regexpr("(~|\\./)?(/)?([^/]*/)*", args[1])) # extract xmloutdir from args[1]
+filename <- basename(args[1]) # Extract file name from args[1]
+
+if(xmloutdir == "") {xmloutdir <- "."}
+
+dbparms = list()
+dbparms$dbname = "bety"
+dbparms$host = "128.197.168.114"
+dbparms$user = "bety"
+dbparms$password = "bety"
+
+#Connection code copied and pasted from met.process
+bety <- dplyr::src_postgres(dbname   = dbparms$dbname, 
+                            host     = dbparms$host, 
+                            user     = dbparms$user, 
+                            password = dbparms$password)
+
+con <- bety$con #Connection to the database.  dplyr returns a list.
+if (is.null(con)) {
+  print("Database connection failed.")
+  quit("no", status=12)
+}
+
+# Set the run dates
+if (exists("source_date")) {
+  start_date <- round.to.six.hours(source_date)
+} else {
+  start_date <- round.to.six.hours()
+}
+
+end_date <- start_date + lubridate::days(16)
+settings$run$start.date <- as.character(start_date)
+settings$run$end.date <- as.character(end_date)
+
+#settings$ensemble$start.year <- start_date
+#settings$ensemble$end.year <- end_date
+
+# Update the time of the run
+settings$info$date <- paste0(format(Sys.time(), "%Y/%m/%d %H:%M:%S"), " +0000")
+
+# Update met.start and met.end for the xml (shouldn't make a difference,
+# but good practice anyway).
+settings$run$site$met.start <- format(Sys.time() - lubridate::days(12), "%Y-%m-%d")
+settings$run$site$met.end <- format(Sys.time() + lubridate::days(16), "%Y-%m-%d")
+
+settings$ensemble$start.year <- format(start_date, "%Y")
+settings$ensemble$end.year <- as.character(end_date, "%Y")
+
+# Determine the workflow ID by looking in the database
+ids <- PEcAn.DB::db.query("SELECT id FROM workflows", con)
+ids <- ids[ids<2000000000] # There are several placeholder ids that are much too large in the database, so this screens
+                           # those out.  Kind of hack-y, but it works.
+workflowid <- as.character(max(ids) + 1)
+settings$workflow$id <- workflowid
+
+#The outdirectory is specific to a particular workflow
+outdir <- settings$outdir
+if (substr(outdir, nchar(outdir), nchar(outdir)) == "/") {
+  outdir <- substr(outdir, 1, nchar(outdir) -1 )
+}
+
+basedir <- regmatches(outdir, regexpr("(~|\\./)?(/)?([^/]*/)*", outdir))
+outdir <- paste0(basedir, "PEcAn_", workflowid, "/")
+settings$outdir <- outdir
+
+# Create the output directory.  PEcAn does not do this for you.  It's normally the job of the
+# web interface's php code.
+if (!dir.exists(outdir)) {
+  dir.create(outdir, recursive=TRUE, showWarnings = FALSE)
+}
+
+# Register new workflow id with the database
+hostname <- PEcAn.remote::fqdn()
+query <- paste0("INSERT INTO workflows (site_id, model_id, notes, folder, hostname, start_date,", 
```

Comment:
This should have a `RETURNING id` at the end and doesn't need the started_at and created_at

Reply:
Looks like all three requested changes have to do with the workflow id/database.  I didn't know the query returned with workflow id after insertion; that's certainly much better.  Not a hard change to make!

Reply:
Workflow ID bad practices have been fixed.